core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -469,14 +469,18 @@ class ReplicaManager(val config: KafkaConfig,
     })
   }
 
-  private def completeDelayedOperationsWhenNotPartitionLeader(topicPartition: TopicPartition, topicId: Option[Uuid]): Unit = {
+  private def completeDelayedOperationsWhenNotPartitionLeader(topicPartition: TopicPartition, topicId: Option[Uuid],
```

Comment:
Perhaps it's clearer to split this into 2 methods, one for NotLeader and another NotReplica?